corehq/apps/data_interfaces/models.py

```diff
@@ -73,36 +73,31 @@ def get_case_ids(cls, domain, case_type, boundary_date=None):
         Retrieves the case ids in chunks, yielding a list of case ids each time
         until there are none left.
         """
-        offset = 0
-        chunk_size = 10000
+        chunk_size = 100
 
-        # We have to sort on _uid; _id is not sortable. See:
-        # https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-uid-field.html
-        # https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-id-field.html
+        query = (CaseES()
+                 .domain(domain)
+                 .case_type(case_type)
+                 .is_closed(closed=False)
+                 .exclude_source()
+                 .size(chunk_size))
 
-        while True:
-            query = (CaseES()
-                     .domain(domain)
-                     .case_type(case_type)
-                     .is_closed(closed=False)
-                     .exclude_source()
-                     .sort('_uid')
-                     .start(offset)
-                     .size(chunk_size))
+        if boundary_date is not None:
+            query = query.server_modified_range(lte=boundary_date)
 
-            if boundary_date is not None:
-                query = query.server_modified_range(lte=boundary_date)
+        result = []
 
-            results = query.run()
-            ids = results.doc_ids
+        for case_id in query.scroll():
```

Comment:
Ah. interesting I missed the yield in the for loop.

I think the timeout is 5 minutes? if you think the processing time wouldbe larger than that, this could be generate a list of lists? I'm not sure what the failure case is with update rules so if it fails gracefully then thsi should be fine